### DISCUSSION TITLE
Default target install tags for shared and static libraries.

### DISCUSSION THREAD

**Discussion by jwidauer** [mesonbuild/meson#12077] [Q&A] [answered]
Hi all,

The [meson docs](https://mesonbuild.com/Installing.html#installation-tags) describe that the default tag for a static library is `devel` and for any other kind of "normal" target (executables, shared libraries, etc.) is `runtime`.
This can also be seen in the implementation [here](https://github.com/mesonbuild/meson/blob/183e4b8e903c6c4c057d09549e669d6292478a4e/mesonbuild/backend/backends.py#L1721C109-L1721C109).

This leads to the behavior that when I run `meson install --tags devel`, I won't actually install any of the shared libraries that are targets in my project.

Is there a particular reason for this behavior? The reason why I believe shared libraries should be tagged both `devel` and `runtime`, is that they are required both when using the library for development, as well as during runtime.
Would this be a worthwhile change to make?

**Comment by xclaesse** [comment, marked as answer]
The intended use-case is for packaging systems that typically will split `libfoo` and `libfoo-dev` where the latter depends on the former. If you want to install both runtime and devel files, simply use `meson install --tags runtime,devel` or just don't give `--tags` argument to install everything.

**Reply by jwidauer** [reply]
Ah, I see! That makes a lot more sense! Thanks for the quick reply!
